### PR TITLE
add rock-3a m.2 e-key sata  dts and defconfig 

### PR DESCRIPTION
**Pull request by pykpkg47**
Added dts for rock-3a which switches m.2 e-key to sata and a defconfig which  enables the  needed options for sata

Tested with m.2 sata ssd , boot works if first  partition(/boot) is vfat 

depends on #94  which has to be merged first

### CODE REVIEW

File: configs/rock-3a-sata-rk3568_defconfig [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+CONFIG_ROCKCHIP_FIT_IMAGE_PACK=y
+CONFIG_ROCKCHIP_NEW_IDB=y
+CONFIG_SPL_SERIAL_SUPPORT=y
+CONFIG_SPL_DRIVERS_MISC_SUPPORT=y
+CONFIG_TARGET_EVB_RK3568=y
+CONFIG_SPL_LIBDISK_SUPPORT=y
+CONFIG_SPL_NAND_SUPPORT=y
+CONFIG_SPL_SPI_FLASH_SUPPORT=y
+CONFIG_SPL_SPI_SUPPORT=y
+CONFIG_DEFAULT_DEVICE_TREE="rk3568-rock-3a-sata"
```
Comment (RadxaStephen) [inline configs/rock-3a-sata-rk3568_defconfig:19]: When you set this variable, you need to kernel rk3568-rock-3a-sata.dtb.

**Comment by nascs** [comment]
Hi @pykpkg47, please add the following configs:

CONFIG_SATA=y 
CONFIG_CMD_SATA=y
CONFIG_SYS_SCSI_MAX_SCSI_ID=1
CONFIG_SYS_SCSI_MAX_LUN=1

**Comment by nascs** [comment]
> Hi @pykpkg47, please add the following configs:
> 
> CONFIG_SATA=y CONFIG_CMD_SATA=y CONFIG_SYS_SCSI_MAX_SCSI_ID=1 CONFIG_SYS_SCSI_MAX_LUN=1

1. you need open sata driver by the config "CONFIG_SATA=y"
2. "CONFIG_CMD_SATA=y" enables this configuration option, which is beneficial for debugging SATA devices on U-Boot

**Comment by nascs** [comment]
> > Hi @pykpkg47, please add the following configs:
> > CONFIG_SATA=y CONFIG_CMD_SATA=y CONFIG_SYS_SCSI_MAX_SCSI_ID=1 CONFIG_SYS_SCSI_MAX_LUN=1
> 
> 1. you need open sata driver by the config "CONFIG_SATA=y"
> 2. "CONFIG_CMD_SATA=y" enables this configuration option, which is beneficial for debugging SATA devices on U-Boot

3. The other two settings are conducive to the more efficient initialization and management of available devices in the system.

**Comment by nascs** [comment]
Certainly, "CONFIG_CMD_SATA=y" is an optional configuration option.

**Comment by pykpkg47** [comment]
added the configs , build u-boot without error, will test later on device

**Comment by pykpkg47** [comment]
Tested with added config options  sata boot still works

**Comment by nascs** [comment]
> Tested with added config options sata boot still works

I tested just now, it works too.

**Comment by RadxaStephen** [comment]
```
diff --git a/configs/rock-3a-sata-rk3568_defconfig b/configs/rock-3a-sata-rk3568_defconfig
new file mode 100644
index 00000000000..dd6cd68fa66
--- /dev/null
+++ b/configs/rock-3a-sata-rk3568_defconfig
@@ -0,0 +1,239 @@
+CONFIG_ARM=y^M
+CONFIG_ARCH_ROCKCHIP=y^M
+CONFIG_SPL_LIBCOMMON_SUPPORT=y^M
+CONFIG_SPL_LIBGENERIC_SUPPORT=y^M
```

@pykpkg47 

Hi. I can see that there is `^M` at the end of each line in file rk3568-rock-3a-sata.dts and rock-3a-sata-rk3568_defconfig.
Please remove the `^M`.

**Comment by pykpkg47** [comment]
done changed  carriage returns to Line Feed (though git in vscode  does that themself but was wrong)

**Comment by RadxaStephen** [comment]
Thanks. I see that rock-3a-sata-rk3568_defconfig is correct. Please also modify file dts rk3568-rock-3a-sata.dts.

**Comment by RadxaStephen** [comment]
Please rewrite these two commits.

1. https://github.com/radxa/u-boot/pull/95/commits/464932b87d1e4b518c09378191b5ce9d088e373e

Remove `^M`  

2. https://github.com/radxa/u-boot/pull/95/commits/fda58fc31ab1d754706b26dd08221a39888b1117

Remove rk3568-rock-3a-sata.dts from this commit since this commit is for rock 3a sata defconfig.

**Comment by RadxaStephen** [comment]
I rewrite the pr and merge it. 

@pykpkg47  Thanks for your great work.

**Pull request closed without merging**